lib/validator.js

```diff
@@ -258,6 +257,46 @@ Specberus.prototype.getSotDSection = function () {
     return this.sotdSection;
 };
 
+Specberus.prototype.extractHeaders = function () {
+  var $ = this.$
+  ,   self = this
+  ,   $dl = $("body div.head dl").first()
+  ,   dts = {}
+  ,   EDITORS_DRAFT = /Latest\ editor's\ draft/i;
+  $dl.find("dt").each(function (idx) {
+    var $dt = $(this)
+    ,   txt = self.norm($dt.text())
+                  .replace(":", "")
+                  .toLowerCase()
+                  .replace("published ", "")
+    ,   $dd = $dt.next('dd')
+    ,   key = null
+    ;
+    if (!$dd) return err("no-dd", { dt: txt });
+    if (txt === "this version") key = "This";
+    else if (txt.lastIndexOf("latest version", 0) === 0) key = "Latest";
+    else if (/^previous version(?:s)?$/.test(txt)) key = "Previous";
+    else if (/^rescinds this recommendation?$/.test(txt)) key = "Rescinds";
+    if (EDITORS_DRAFT.test(txt) && $dd.find('a')) key = "Editor";
+    if (key) dts[key] = { pos: idx, el: $dt, dd: $dd };
+  });
+  return dts;
+};
+
+Specberus.prototype.getEditorIDs = function () {
```

Comment:
Shouldn't the logic of this function be inside `lib/rules/metadata/editor-ids.js` instead?
`lib/validator.js` can store its output locally when it runs, and later serve that value to other rules that need it (eg `lib/rules/echidna/editor-ids.js`).

Two reasons: keep all metadata extraction under `lib/rules/metadata/`, and scan the document in search of editor IDs only once (as it is now, it runs twice, iiuc).